### PR TITLE
Move nft e2e tests to main e2e test job

### PR DESCRIPTION
**Pull request by PeterYinusa**
## Explanation

This PR removes the test nft builds and e2e nft jobs, and moves the nft tests into the main e2e test job 
## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by PeterYinusa** [comment]
This change should either be merged into https://github.com/MetaMask/metamask-extension/pull/17401 or we can change the base branch to `develop` once the other PR is merged

File: package.json [hunk 1]
```diff
@@ -32,10 +32,8 @@
     "test:unit:mocha": "node ./test/run-unit-tests.js --mocha",
     "test:e2e:chrome": "SELENIUM_BROWSER=chrome node test/e2e/run-all.js",
     "test:e2e:chrome:snaps": "SELENIUM_BROWSER=chrome node test/e2e/run-all.js --snaps",
-    "test:e2e:chrome:nft": "SELENIUM_BROWSER=chrome node test/e2e/run-all.js --nft",
```
Comment (PeterYinusa) [inline package.json:35]: NFT tests now ran as part of `test:e2e:chrome`

File: package.json [hunk 2]
```diff
@@ -32,10 +32,8 @@
...
     "test:e2e:firefox": "SELENIUM_BROWSER=firefox node test/e2e/run-all.js",
     "test:e2e:firefox:snaps": "SELENIUM_BROWSER=firefox node test/e2e/run-all.js --snaps",
-    "test:e2e:firefox:nft": "SELENIUM_BROWSER=firefox node test/e2e/run-all.js --nft",
```
Comment (PeterYinusa) [inline package.json:38]: NFT tests now ran as part of `test:e2e:firefox`

File: test/e2e/nft/nft.spec.js [hunk 3]
```diff
@@ -1,33 +0,0 @@
-const { strict: assert } = require('assert');
```
Comment (PeterYinusa) [inline test/e2e/nft/nft.spec.js:1]: This test was to ensure we was testing a build with the NFT feature enabled, this is no longer needed as this feature is in the main build and no longer behind a feature flag

File: test/e2e/run-all.js [hunk 4]
```diff
@@ -63,21 +59,19 @@ async function main() {
...
     testPaths = await getTestPathsForTestDir(testDir);
-  } else if (nft) {
-    const testDir = path.join(__dirname, 'nft');
-    testPaths = await getTestPathsForTestDir(testDir);
   } else {
     const testDir = path.join(__dirname, 'tests');
     testPaths = [
       ...(await getTestPathsForTestDir(testDir)),
       ...(await getTestPathsForTestDir(path.join(__dirname, 'swaps'))),
+      ...(await getTestPathsForTestDir(path.join(__dirname, 'nft'))),
```
Comment (PeterYinusa) [inline test/e2e/run-all.js:74]: The NFT tests should be run in the main e2e test run, alongside all the other specs

**Review by brad-decker — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM !

**Pull request merged**